src/sign/types.ts

```diff
@@ -0,0 +1,13 @@
+export interface Subscription {
+  address: string;
+  email: string;
+  action: string;
+}
+
+export const SubscriptionTypes = {
+  Subscriber: [
+    { name: 'address', type: 'address' },
+    { name: 'email', type: 'string' },
+    { name: 'action', type: 'string' }
+  ]
```

Comment:
Would be better to have a message type for each actions like this:
```suggestion
  Subscribe: [
    { name: 'address', type: 'address' },
    { name: 'email', type: 'string' }
  ]
```